src/libraries/Common/src/Roslyn/GetBestTypeByMetadataName.cs

```diff
@@ -134,5 +138,40 @@ private enum SymbolVisibility
             Private = 2,
             Friend = Internal,
         }
+
+        internal static bool HasAttributeSuffix(this string name, bool isCaseSensitive)
+        {
+            const string AttributeSuffix = "Attribute";
+
+            var comparison = isCaseSensitive ? StringComparison.Ordinal : StringComparison.OrdinalIgnoreCase;
+            return name.Length > AttributeSuffix.Length && name.EndsWith(AttributeSuffix, comparison);
+        }
+
+        public static ImmutableArray<T> ToImmutableArray<T>(this ReadOnlySpan<T> span)
+        {
+            switch (span.Length)
+            {
+                case 0: return ImmutableArray<T>.Empty;
+                case 1: return ImmutableArray.Create(span[0]);
+                case 2: return ImmutableArray.Create(span[0], span[1]);
+                case 3: return ImmutableArray.Create(span[0], span[1], span[2]);
+                case 4: return ImmutableArray.Create(span[0], span[1], span[2], span[3]);
+                default:
+                    var builder = ImmutableArray.CreateBuilder<T>(span.Length);
```

Comment:
NIT: we typically try to use explicit types unless the exact type can be inferred from the right-side of the assignment.

Reply:
Please don't make me do this. The more we deviate from Roslyn, the harder this will be to support.

Reply:
Sounds fair